src/macro_util.rs

```diff
@@ -500,6 +500,74 @@ where
     Ok(unsafe { mem::transmute_copy(&*src) })
 }
 
+/// Attempts to transmute `&Src` into `&Dst`.
+///
+/// A helper for `try_transmute_ref!`.
+///
+/// # Panics
+///
+/// `try_transmute_ref` may either produce a post-monomorphization error or a
+/// panic if `Dst` is bigger than `Src`. Otherwise, `try_transmute_ref` panics
+/// under the same circumstances as [`is_bit_valid`].
+///
+/// [`is_bit_valid`]: TryFromBytes::is_bit_valid
+#[inline(always)]
+pub fn try_transmute_ref<'a, Src, Dst>(src: &'a Src) -> Result<&'a Dst, ValidityError<&'a Src, Dst>>
+where
+    Src: IntoBytes + Immutable,
+    Dst: TryFromBytes + Immutable,
+{
+    if !is_src_valid::<Src, Dst, _, BecauseImmutable>(Ptr::from_ref(&src)) {
+        return Err(ValidityError::new(src));
+    }
+
+    let src = ManuallyDrop::new(src);
+
+    // SAFETY: By contract on `is_src_valid`, we have confirmed both that `Dst`
+    // is no larger than `Src`, and that `src` is a bit-valid instance of `Dst`.
+    // These conditions are preserved through the `ManuallyDrop<T>` wrapper,
+    // which is documented to have identical and layout bit validity to its
+    // inner value [1].
+    //
+    // [1]: https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html
+    Ok(unsafe { mem::transmute_copy(&*src) })
```

Comment:
The `transmute_copy` here is overly awkward (I'll fix that), but the bigger issue is the `Sized` bound. If we retain it, we can keep the static size errors. If we ditch it, we can handle DSTs (but size errors for all types will be dynamic, not static). What should we do?

Reply:
perhaps we support both:
- `try_transmute_ref!(src)` with return type `Result<Dst, ValidityError<&Src, Dst>>` and static size/alignment checking
- `try_transmute_ref!(unsized src)` with return type `Result<Dst, TryCastError<&Src, Dst>>` (dynamic size/alignment checking), but support for DSTs.